### PR TITLE
Improve block processing tests

### LINKED ISSUE #8135: Unit test review for block processing (LIP 0055)
## Block

- Some more precise tests regarding `MAX_TRANSACTIONS_SIZE_BYTES` than the [exiting one](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/chain.spec.ts#L318) would be good. That means, one test where the size equals `MAX_TRANSACTIONS_SIZE_BYTES` and that passes. And one test where the size equals `MAX_TRANSACTIONS_SIZE_BYTES+1` and that fails.
- It should be tested that [`assets.validate()`](https://github.com/LiskHQ/lisk-sdk/blob/de09b745a369c0970db7a8db2e5980534a1a036a/elements/lisk-chain/src/block_assets.ts#L86) and [`blockHeader.validate()`](https://github.com/LiskHQ/lisk-sdk/blob/de09b745a369c0970db7a8db2e5980534a1a036a/elements/lisk-chain/src/block_header.ts#L181) are called during block processing and that block processing fails if one of them fails.
- It should be tested that for every transaction in the block, every hook ("Transaction verification", ..., "After command execution") is executed and that the whole block is discarded if one of the hook calls fails (except for "Command execution") .

## Block Header

- The properties `eventRoot` and `impliesMaxPrevotes` are missing in `blockHeaderProps` in [block_header.spec.ts](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_header.spec.ts#L77).
- The test for the [constructor](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_header.spec.ts#L130) misses the check for the `impliesMaxPrevotes` property.

### Timestamp

- Test for "timestamp in the future": 
	- It seems that [this line](https://github.com/LiskHQ/lisk-sdk/blob/b009c9f7aef93464eb82823cfa833116b7eb2d3d/framework/test/unit/engine/consensus/consensus.spec.ts#L734) assigns the same slot number to the timestamp of the block, the current system time and the timestamp of the previous block, which should not be the case. So, an error is thrown because the slots of the block and the previous block are equal, and not because the timestamp is too far in the future.
	- Also, the slot returned by the mocked function should be `Math.floor(Date.now() / 10000)` instead of `Math.floor(Date.now() / 10)`?

	
### State Root

- There is no test for the `stateRoot` property. In particular, there is no test that expects that block processing fails if the value of `header.stateRoot` in incorrect.

### BFT properties

- test for invalid `impliesMaximalPrevotes` property is missing
- in the test for the [success case](https://github.com/LiskHQ/lisk-sdk/blob/de09b745a369c0970db7a8db2e5980534a1a036a/framework/test/unit/engine/consensus/consensus.spec.ts#L872), also the function `impliesMaximalPrevotes` should be mocked to ensure that it returns true.

## Block Assets

- Test for [exceeding `MAX_ASSET_DATA_SIZE_BYTES`](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_assets.spec.ts#L128): It would be good if the size of the `data` property equals `MAX_ASSET_DATA_SIZE_BYTES+1` instead of an arbitrary value.
- Also for the test for [not exceeding the size](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_assets.spec.ts#L145), it may be better to use `MAX_ASSET_DATA_SIZE_BYTES` instead of 64 to make it independent of the value of `MAX_ASSET_DATA_SIZE_BYTES`. But this is not so important.
- [test for non-sorted assets](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_assets.spec.ts#L161): it is still talking about the `moduleId` instead of `module`. Also in the [expectation](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_assets.spec.ts#L175).
  - Also [here](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_assets.spec.ts#L195) and [here](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/block_assets.spec.ts#L213).
- There seems to be no test to ensure that block processing fails if `asset.module` does not refer to a registered module for any asset `asset` in the block.

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #8135 

### How was it solved?

- Added missing unit tests

### How was it tested?

- All the missing tests the issue mentioned is added except the one replied in the comment of #8135

### CODE REVIEW

**Review by AndreasKendziorra — CHANGES_REQUESTED** [review]
There are still some points in #8135 that are in discussion.
Moreover, the following point from #8135 was not addressed:

> Some more precise tests regarding MAX_TRANSACTIONS_SIZE_BYTES than the [exiting one](https://github.com/LiskHQ/lisk-sdk/blob/42897f79e56bcf28a876899a91f73cb98c0a7932/elements/lisk-chain/test/unit/chain.spec.ts#L318) would be good. That means, one test where the size equals MAX_TRANSACTIONS_SIZE_BYTES and that passes. And one test where the size equals MAX_TRANSACTIONS_SIZE_BYTES+1 and that fails.

File: elements/lisk-chain/src/block_assets.ts [hunk 1]
```diff
@@ -100,7 +100,9 @@ export class BlockAssets {
 				);
 			}
 			if (last.module > asset.module) {
-				throw new Error('Assets are not sorted in the increasing values of moduleID.');
+				throw new Error(
+					'Assets Assets are not sorted by the module property value in lexicographical order.',
```
Comment (Incede) [inline elements/lisk-chain/src/block_assets.ts:104]:
```suggestion
					'Assets are not sorted by the module property value in lexicographical order.',
```

**Review by Incede — APPROVED** [review]

**Comment by AndreasKendziorra** [comment]
I just saw that I falsely left my [question](https://github.com/LiskHQ/lisk-sdk/issues/8135#issuecomment-1547401667) regarding `MAX_TRANSACTIONS_SIZE_BYTES` in the issue instead of here. I would do the next review once this discussion is settled or addressed.

**Comment by shuse2** [comment]
@AndreasKendziorra I updated the chain test for the payload size test in https://github.com/LiskHQ/lisk-sdk/pull/8444/commits/621d94a595a5b01d2e98c7a49c8c729b32673fb7

**Review by AndreasKendziorra — APPROVED** [review]

**Pull request merged**